src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Helpers.cs

```diff
@@ -128,7 +128,7 @@ public static X509Certificate2 TryFindMatchingCertificate(this X509Certificate2C
 
         private static bool AreByteArraysEqual(byte[] ba1, byte[] ba2)
         {
-            if (ba1.Length != ba2.Length)
+            if (ba1 == null || ba2 == null || ba1.Length != ba2.Length)
```

Comment:
This is slightly incorrect.  if they're both null then they would be equal.  Or, since this method wasn't expecting null, does it instead make sense to check for null at the caller?